src/Google/GlobalSiteTag.php

```diff
@@ -78,6 +78,13 @@ function() use ( $ads_conversion_id, $ads_conversion_label ) {
 			},
 			1000000
 		);
+		add_action(
+			'wp_body_open',
```

Comment:
The hook `wp_body_open` has been added since WordPress 5.2, but it relies on the themes having added this call in their templates. We might be missing out some tracking on sites that use an older theme still.

Also this hook means we are going to run the function for every single page, most of which we exit early because `is_product` is not true. However in our Google Analytics extension we use hooks which are called specifically for the single product page, like `woocommerce_after_single_product`. If that's not early enough the hook `woocommerce_before_single_product` can also be used.